packages/react/src/ReactElementValidator.js

```diff
@@ -281,7 +285,7 @@ export function createElementWithValidation(type, props, children) {
   const validType =
     typeof type === 'string' ||
     typeof type === 'function' ||
-    typeof type === 'symbol' ||
+    isReactSymbol(type) ||
```

Comment:
I think we can leave this check as is. It’s a hot path so we can’t do `.some()` there. It’s also unnecessary because the number check below should cover it.